app/main.py

```diff
@@ -1,3 +1,65 @@
-def shop_trip():
-    # write your code here
-    pass
+from datetime import datetime
+import json
+
+from app.shop import Shop
+from app.customer import Customer
+
+
+def shop_trip() -> None:
+    with open("config.json") as config:
+        config_data = json.load(config)
+    fuel_prize = config_data["FUEL_PRICE"]
+
+    list_customers = [Customer(customers["name"],
+                               customers["product_cart"],
+                               customers["location"],
+                               customers["money"],
+                               customers["car"]
+                               ) for customers in config_data["customers"]]
+
+    list_shops = [Shop(
+        shops["name"],
+        shops["location"],
+        shops["products"])
+        for shops in config_data["shops"]]
+
+    for customer in list_customers:
+        customer.customer_has_money()
+        costs = float("inf")
+        shop_obj = None
+        for shop in list_shops:
+            min_cost = (customer.cost_of_the_trip(
+                fuel_prize,
+                customer.location,
+                shop.location) * 2 + shop.prise_products(
```

Comment:
The method `prise_products` is likely a typo and should be `price_products` based on its functionality. Ensure that the method name matches the intended purpose.